### PR TITLE
Prefer page path methods for start/summary

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR adds new page controller getters/methods to assist when walking the form journey

* `page.href`
* `page.getStartPath()`

Mainly to avoid this everywhere:

```patch
- return `/${this.model.basePath || ''}${page.path}`
+ return page.href
```

I've also simplified some page methods where possible

## Test tweaks

I've also added `basePath` to form tests and removed the consts we added to appease Sonar:

```patch
- const okStatusCode = 200
- const redirectStatusCode = 302
- const badRequestStatusCode = 400
```

### CODE REVIEW

File: src/server/index.test.ts [hunk 1]
```diff
@@ -10,10 +10,6 @@ import * as fixtures from '~/test/fixtures/index.js'
 
 jest.mock('~/src/server/plugins/engine/services/formsService.js')
 
-const okStatusCode = 200
-const redirectStatusCode = 302
-const notFoundStatusCode = 404
```
Comment (mokhld) [inline src/server/index.test.ts:15]: @colinrotherham out of interest, why did you remove this? Seems to make sense to keep it DRY?

Comment (mokhld) [reply src/server/index.test.ts:15]: Sorry ignore my question. I saw your comment about Sonar being the culprit.

Comment (colinrotherham) [reply src/server/index.test.ts:15]: Yeah agreed. We should swap them all for `StatusCodes` at some point

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts [hunk 2]
```diff
@@ -169,6 +167,11 @@ export class PageControllerBase {
     }
   }
 
+  get href() {
+    const { model, path } = this
+    return `/${model.basePath}${path}`
```
Comment (mokhld) [inline src/server/plugins/engine/pageControllers/PageControllerBase.ts:172]: Is there ever a case where we might have malformed URLs?

**Review by mokhld — APPROVED** [review]

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts (hunk 2 above)
Comment (colinrotherham) [reply src/server/plugins/engine/pageControllers/PageControllerBase.ts:172]: Possibly as we support form definitions with `startPage` starting `http(s)://`

Not due to this PR, but we may have potential redirect issues lurking somewhere

**Review by mokhld — APPROVED** [review]
